.github/workflows/perf.yml

```diff
@@ -0,0 +1,69 @@
+name: perf
+
+# build on c/cpp changes or workflow changes
+on:
+  push:
+    paths:
+      - 'lib/**.[ch]'
+      - 'lib/**.cpp'
+      - 'src/**.[ch]'
+      - 'src/**.cpp'
+      - '**/CMakeLists.txt'
+      - 'cmake/Modules/**'
+      - 'util/buildbot/**'
+      - 'util/ci/**'
+      - '.github/workflows/**.yml'
+      - 'Dockerfile'
+      - '.dockerignore'
+# IMPORTANT: Do not run this workflow on pull request since this workflow
+# has permission to modify contents.
+
+permissions:
+  # deployments permission to deploy GitHub pages website
+  deployments: write
+  # contents permission to update benchmark contents in gh-pages branch
+  contents: write
+
+jobs:
+  # Current gcc version
+  perf_gcc_10:
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Install deps
+        run: |
+          source ./util/ci/common.sh
+          install_linux_deps g++-10
+
+      - name: Build
+        run: |
+          ./util/ci/build.sh
```

Comment:
Actually this builds a debug build which is bad for benchmarking, you could make this change
```diff
diff --git a/util/ci/build.sh b/util/ci/build.sh
index 435cc11da..a10080ca1 100755
--- a/util/ci/build.sh
+++ b/util/ci/build.sh
@@ -1,6 +1,6 @@
 #! /bin/bash -e
 
-cmake -B build -DCMAKE_BUILD_TYPE=Debug \
+cmake -B build -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE:-Debug} \
        -DRUN_IN_PLACE=TRUE -DENABLE_GETTEXT=TRUE \
        -DBUILD_SERVER=TRUE ${CMAKE_FLAGS}
 cmake --build build --parallel $(($(nproc) + 1))
```
and then set the right env var to get a release build

Reply:
done